crates/build/src/manifest.rs

```diff
@@ -4,37 +4,55 @@ use std::{collections::BTreeMap, path::Path};
 
 use spin_manifest::{schema::v2, ManifestVersion};
 
+pub type DeploymentTarget = String;
+pub type DeploymentTargets = Vec<DeploymentTarget>;
```

Comment:
I think if we switch `component_build_configs` to returning a more descriptive type, than we can get rid of these type aliases. I personally find type aliases that alias core types like `String` and `Vec` more confusing than helpful.